### ISSUE TITLE
Weird data type conversion happens for `int32` and `uint32`

### ISSUE THREAD

**Issue by hanjinliu** [pola-rs/polars#6559]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

Numpy arrays after `%` or `//` are converted into `Object`-type data frames.

### Reproducible example

```python
for dtype in [np.uint8, np.uint16, np.uint32, np.uint64, np.float16, np.float32, np.float64, np.int32, np.int64]:
    arr = np.array([2,1,3], dtype=dtype)
    print(pl.DataFrame(arr).dtypes, pl.DataFrame(arr % 2).dtypes)
```

### Expected behavior

Should be `Int32`/`Uint32`.

### Installed versions

<details>

```
---Version info---
Polars: 0.16.1
Index type: UInt32
Platform: Windows-10-10.0.22621-SP0
Python: 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:16) [MSC v.1916 64 bit (AMD64)]
---Optional dependencies---
pyarrow: 10.0.0
pandas: 1.5.2
numpy: 1.23.2
fsspec: 2021.11.0
connectorx: <not installed>
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: 3.5.3
```

</details>

**Comment by ritchie46** [comment]
This is my output:

```python
for dtype in [np.uint8, np.uint16, np.uint32, np.uint64, np.float16, np.float32, np.float64, np.int32, np.int64]:
    arr = np.array([2,1,3], dtype=dtype)
    print(pl.DataFrame(arr).dtypes, pl.DataFrame(arr % 2).dtypes, dtype)
```
```
[UInt8] [UInt8] <class 'numpy.uint8'>
[UInt16] [UInt16] <class 'numpy.uint16'>
[UInt32] [UInt32] <class 'numpy.uint32'>
[UInt64] [UInt64] <class 'numpy.uint64'>
[Float32] [Float32] <class 'numpy.float16'>
[Float32] [Float32] <class 'numpy.float32'>
[Float64] [Float64] <class 'numpy.float64'>
[Int32] [Int32] <class 'numpy.int32'>
[Int64] [Int64] <class 'numpy.int64'>
```

**Comment by hanjinliu** [comment]
Hi @ritchie46 ,
thank you for your prompt response.

Here's my output:

![image](https://user-images.githubusercontent.com/40591297/215428292-faf2b3bc-7b87-45d4-ad98-60e97b7138be.png)

**Comment by ritchie46** [comment]
And from this?

```
print((arr % 2).dtype, dtype)
```

**Comment by hanjinliu** [comment]
It returns correct results.

![image](https://user-images.githubusercontent.com/40591297/215429565-1c22e9b4-1617-4d0f-a2bd-751564f16e78.png)

**Comment by ghuls** [comment]
Could you try to update numpy? Quite unlikely it will solve it, but you never know.

What is the output of the following?
```python
for dtype in [np.uint8, np.uint16, np.uint32, np.uint64, np.float16, np.float32, np.float64, np.int32, np.int64]:
   arr = np.array([2,1,3], dtype=dtype)
   arr_2 = arr % 2
   print(arr.dtype, arr.dtype.char, arr_2.dtype, arr_2.dtype.char, pl.DataFrame(arr).dtypes, pl.DataFrame(arr % 2).dtypes, pl.DataFrame((arr % 2)).dtypes, pl.DataFrame(arr_2).dtypes)
```

**Comment by hanjinliu** [comment]
Updating `numpy` didn't change anything.

I ran your example and got output like below.

```
uint8 B uint8 B [UInt8] [UInt8] [UInt8] [UInt8]
uint16 H uint16 H [UInt16] [UInt16] [UInt16] [UInt16]
uint32 L uint32 I [UInt32] [Object] [Object] [Object]
uint64 Q uint64 Q [UInt64] [UInt64] [UInt64] [UInt64]
float16 e float16 e [Float32] [Float32] [Float32] [Float32]
float32 f float32 f [Float32] [Float32] [Float32] [Float32]
float64 d float64 d [Float64] [Float64] [Float64] [Float64]
int32 l int32 i [Int32] [Object] [Object] [Object]
int64 q int64 q [Int64] [Int64] [Int64] [Int64]
```

**Comment by ghuls** [comment]
It looks like on Windows the char code associated with the numpy dtype uint32 and int32 changes for `arr_2 = arr % 2` : `uint32 L uint32 I` and `int32 l int32 i`

**Comment by ghuls** [comment]
With a lot of luck, it might have been fixed by https://github.com/pola-rs/polars/pull/6531, if this is problem was caused by the rust numpy bindings itself.

**Comment by ghuls** [comment]
Working on a solution.

**Comment by ghuls** [comment]
Closed by #6738.
